README.md

```diff
@@ -166,6 +162,8 @@ Available configuration options:
 * `iterationTimeUnit` – time unit for `iterationTime` (default is seconds)
 * `outputTimeUnit` – time unit for results output
 * `mode` – "thrpt" for measuring operations per time, or "avgt" for measuring time per operation
+* `iterationMode` – "external" for iterating in gradle in order to get correct Kotlin/Native runtime input in measurement,
+ "internal" can be used if it's known that measured code have no calls in K/N runtime that can influence on measurement unrepeatedly.
```

Comment:
From this description it's very hard to get when one should use this configuration option. Perhaps, if there's a sensible default, we could turn it into a boolean option that would allow to override this default and explain why users would want to do it.

Reply:
"External" is already such default. It's similar to other settings that also have. default values, but about description I agree. If you have ideas what to write. in README, I'll be greatful

Reply:
`* iterationMode – way of iteration for K/N benchmarks. Default value - "external", each iteration is processed as separate run, allows to get more precise results in most cases. "internal" - all iterations of one benchmark run during one binary execution (pay attention, some code performance can't be measured properly using this mode, e.g. singleton initialization, because of implementation details of K/N runtime)`

Is this better?

Reply:
It feels conceptually like 'fork' option in JMH (see the [example](https://github.com/openjdk/jmh/blob/master/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_12_Forking.java)). Perhaps we could find a resembling name for this option.

Reply:
> singleton initialization

I always thought that microbenchmarks tried to measure the settled code execution performance, i.e. when all fixed initialization costs are done and no longer affect the performance of the code being measured. That's why they do warm-up iterations.
So for me it looks like that the mode of a fork-per-benchmark method should be the preferred default.

Reply:
I thought during implementation about fork, but it's different in jmh. Firstly fork is number and user can set by user. Also as I undertood JMH created needed number of forks and warmups and iterations is made inside each fork, logic of working for "external" mode is another, forks are even more similar for "internal" mode where is one fork for each benchmark. I really think that using `fork` can confuse users, because they'll expect the same behaviour for Native.

Reply:
> when all fixed initialization costs are done and no longer affect the performance of the code being measured.

It isn't true for native runtime. Done initialization may influence on the next code. And we don't know what exactly user want to measure. 

> That's why they do warm-up iterations.

Warm-uo for native applications is needed to warmup processors caches, etc..

> So for me it looks like that the mode of a fork-per-benchmark method should be the preferred default.

Any code can influence on thresholds inside runtime and that can influence on performance, if user doesn't know runtime implementation details and want to get accurate results he should use "external". We can change default, but I amn't sure that it's right for majority of users. May be we should discuss this, it was already discussed with @qurbonzoda